### PR TITLE
fix prev chunk size check

### PR DESCRIPTION
**Pull request by lonnywong**
**before fix :**
image

**after fix :**
image

### CODE REVIEW

**Comment by disconnect3d** [comment]
@GrosQuildu can you check that one?

**Comment by disconnect3d** [comment]
@lonnywong can you give some more context? it's hard to judge if the  change is correct from the screenshots :P

**Comment by lonnywong** [comment]
> @lonnywong can you give some more context? it's hard to judge if the change is correct from the screenshots :P

```c
#include <stdio.h>
#include <stdlib.h>

int main() {
	void *p1 = malloc(0x500);
	void *p2 = malloc(0x500);
	free(p1);
	printf("pwndbg> try_free %p\n", p2);
	getchar();
	return 0;
}
```

File: pwndbg/commands/heap.py [hunk 1]
```diff
@@ -932,7 +932,7 @@ def finalize(errors_found, returned_before_error):
                 finalize(errors_found, returned_before_error)
                 return
 
-            if unsigned_size(prev_chunk['size']) != prev_size:
+            if chunksize(unsigned_size(prev_chunk['size'])) != prev_size:
```
Comment (GrosQuildu) [inline pwndbg/commands/heap.py]:
```suggestion
            if prev_chunk_size != prev_size:
```

File: pwndbg/commands/heap.py [hunk 2]
```diff
@@ -932,7 +932,7 @@ def finalize(errors_found, returned_before_error):
...
                 err = 'corrupted size vs. prev_size while consolidating\n'
                 err += 'prev_size field is 0x{:x}, prev chunk at 0x{:x}, prev chunk size is 0x{:x}'
                 err = err.format(prev_size, prev_chunk_addr, unsigned_size(prev_chunk['size']))
```
Comment (GrosQuildu) [inline pwndbg/commands/heap.py]:
```suggestion
                err = err.format(prev_size, prev_chunk_addr, prev_chunk_size)
```

**Review by GrosQuildu — CHANGES_REQUESTED** [review]
Cool, thanks. We may simplify it and merge.

File: pwndbg/commands/heap.py (hunk 1 above)
Comment (lonnywong) [reply pwndbg/commands/heap.py]: Good point.

File: pwndbg/commands/heap.py (hunk 2 above)
Comment (lonnywong) [reply pwndbg/commands/heap.py]: Good point.

**Review by GrosQuildu — APPROVED** [review]

**Comment by disconnect3d** [comment]
Thanks!

**Pull request merged**